runtime/hydradx/src/benchmarking/omnipool_liquidity_mining.rs

```diff
@@ -0,0 +1,969 @@
+		Currencies::update_balance(
+			RawOrigin::Root.into(),
+			acc.clone(),
+			asset_id,
+			(token_amount * 100) as Amount,
+		)?;
+		Omnipool::add_token(
+			RawOrigin::Root.into(),
+			asset_id,
+			token_price,
+			Permill::from_percent(100),
+			owner,
+		)?;
+	}
+
+	//NOTE: This is necessary for oracle to provide price.
+	set_period(10);
+
+	do_lrna_hdx_trade()
+}
+
+//NOTE: This is necessary for oracle to provide price.
+fn do_lrna_hdx_trade() -> DispatchResult {
+	let trader = funded_account("tmp_trader", 0, &[REWARD_CURRENCY]);
+
+	fund(trader.clone(), LRNA, 100 * ONE)?;
+
+	Omnipool::sell(RawOrigin::Signed(trader).into(), LRNA, HDX, ONE, 0)
+}
+
+fn seed_lm_pot() -> DispatchResult {
+	let pot = OmnipoolWarehouseLM::pot_account_id().unwrap();
+
+	fund(pot, HDX, 100 * ONE)
+}
+
+fn omnipool_add_liquidity(lp: AccountId, asset: AssetId, amount: Balance) -> Result<u128, DispatchError> {
+	let current_position_id = Omnipool::next_position_id();
+
+	Omnipool::add_liquidity(RawOrigin::Signed(lp).into(), asset, amount)?;
+
+	Ok(current_position_id)
+}
+
+fn lm_deposit_shares(who: AccountId, g_id: GlobalFarmId, y_id: YieldFarmId, position_id: ItemId) -> DispatchResult {
+	OmnipoolLiquidityMining::deposit_shares(RawOrigin::Signed(who).into(), g_id, y_id, position_id)
+}
+
+fn set_period(to: u32) {
+	//NOTE: predefined global farm has period size = 1 block.
+	while System::block_number() < to {
+		let b = System::block_number();
+
+		<pallet_circuit_breaker::Pallet<Runtime> as OnFinalize<BlockNumberFor<crate::Runtime>>>::on_finalize(b);
+		<frame_system::Pallet<Runtime> as OnFinalize<BlockNumberFor<crate::Runtime>>>::on_finalize(b);
+		<pallet_ema_oracle::Pallet<Runtime> as frame_support::traits::OnFinalize<BlockNumberFor<crate::Runtime>>>::on_finalize(b);
+
+		<pallet_circuit_breaker::Pallet<Runtime> as OnInitialize<BlockNumberFor<crate::Runtime>>>::on_initialize(
+			b + 1_u32,
+		);
+		<frame_system::Pallet<Runtime> as OnInitialize<BlockNumberFor<crate::Runtime>>>::on_initialize(b + 1_u32);
+		<pallet_ema_oracle::Pallet<Runtime> as frame_support::traits::OnInitialize<BlockNumberFor<crate::Runtime>>>::on_initialize(b + 1_u32);
+
+		System::set_block_number(b + 1_u32);
+	}
+}
+
+runtime_benchmarks! {
+	{Runtime, pallet_omnipool_liquidity_mining }
+
+	create_global_farm {
+		let planned_yielding_periods = BlockNumberFor::<crate::Runtime>::from(100_000_u32);
+		let blocks_per_period = BlockNumberFor::<crate::Runtime>::from(100_u32);
+		let reward_currency = register_asset(b"REW".to_vec(), 10000 * ONE).map_err(|_| BenchmarkError::Stop("Failed to register asset"))?;
+		let owner = funded_account("owner", 0, &[REWARD_CURRENCY]);
+		let yield_per_period = Perquintill::from_percent(20);
+		let min_deposit = 1_000;
+		let price_adjustment = FixedU128::from(10_u128);
+
+	}: _(RawOrigin::Root,  G_FARM_TOTAL_REWARDS, planned_yielding_periods, blocks_per_period, REWARD_CURRENCY, owner, yield_per_period, min_deposit, FixedU128::one())
+
+	update_global_farm {
```

Comment:
This benchmark is not correct. `update_global_farm` is doing more work when farm is not empty